core/src/main/java/tc/oc/pgm/action/ActionParser.java

```diff
@@ -440,4 +441,9 @@ public <T extends Filterable<?>> PasteStructureAction<T> parseStructure(
 
     return new PasteStructureAction<>(scope, xFormula, yFormula, zFormula, structure, update);
   }
+
+  @MethodParser("weather")
+  public WeatherAction parseWeather(Element el, Class<?> scope) throws InvalidXMLException {
+    return new WeatherAction(XMLUtils.parseEnum(Node.fromAttr(el, "state"), WeatherType.class));
```

Comment:
use the new fluent parsing, it makes the code clearer, as well as fixing the issue that this current code would allow a null (ie: it is not a required attribute) which would then error out when trying to run the weather action:
```suggestion
    return new WeatherAction(parser.parseEnum(WeatherType.class, el, "style").required());
```